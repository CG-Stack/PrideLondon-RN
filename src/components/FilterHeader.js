// @flow
import React from "react";
import {
  View,
  StyleSheet,
  StatusBar,
  TouchableOpacity,
  Image
} from "react-native";
import SafeAreaView from "react-native-safe-area-view";
import DateFilterDialog from "./ConnectedDateFilterDialog";
import FilterHeaderButton from "./FilterHeaderButton";
import TimeFilterDialog from "./ConnectedTimeFilterDialog";
import Text from "./Text";
import ContentPadding from "./ContentPadding";
import {
  interestButtonBgColor,
  interestButtonTextColor,
  filterBgColor,
  filterShowMeTextColor,
  categoriesFilterButtonBgColor
} from "../constants/colors";
import text from "../constants/text";
import type { DateOrDateRange, Time } from "../data/date-time";
import { formatDateRange } from "../data/formatters";

import chevronRightImg from "../../assets/images/chevronRight.png";

type Props = {
  dateFilter: ?DateOrDateRange,
  timeFilter: Set<Time>
};

type State = {
  datesPickerVisible: boolean,
  timesPickerVisible: boolean
};

class FilterHeader extends React.PureComponent<Props, State> {
  state = {
    datesPickerVisible: false,
    timesPickerVisible: false
  };

  showDatePicker = () => {
    this.setState({ datesPickerVisible: true });
  };

  hideDatePicker = () => {
    this.setState({ datesPickerVisible: false });
  };

  showTimePicker = () => {
    this.setState({ timesPickerVisible: true });
  };

  hideTimePicker = () => {
    this.setState({ timesPickerVisible: false });
  };

  render() {
    const { dateFilter, timeFilter } = this.props;
    const formattedDateFilter = dateFilter
      ? formatDateRange(dateFilter)
      : text.anyDay;
    const timeArray = Array.from(timeFilter);
    const formattedTimeFilter =
      timeArray.length > 0 && timeArray.length < 3
        ? timeArray.map(time => text.time[time]).join(", ")
        : text.anyTime;

    return (
      <SafeAreaView style={styles.container} forceInset={{ top: "always" }}>
        <StatusBar barStyle="light-content" animated />
        <ContentPadding>
          <View testID="filter-header" style={styles.content}>
            <View style={styles.contentInterest}>
              <Text type="h1" style={styles.filterTitle}>
                {text.filterTitle}
              </Text>
              <View style={styles.interestButton}>
                <Text type="h2" style={styles.interestButtonText}>
                  {text.filterByInterest}
                </Text>
                <TouchableOpacity
                  accessibilityTraits={["button"]}
                  accessibilityComponentType="button"
                  style={styles.categoriesFilterButton}
                >
                  <Image source={chevronRightImg} />
                </TouchableOpacity>
              </View>
              <View style={styles.mapButton}>
                <Text style={styles.mapButtonText}>Map</Text>
              </View>
            </View>
            <View style={styles.contentFilters}>
              <FilterHeaderButton
                text={formattedDateFilter}
                onPress={this.showDatePicker}
                style={styles.filterButton}
              />
              <FilterHeaderButton
                text={formattedTimeFilter}
                onPress={this.showTimePicker}
                style={styles.filterButton}
              />
              <FilterHeaderButton
                text={text.filters}
                onPress={() => {}}
                style={styles.filterButton}
              />
            </View>
          </View>
        </ContentPadding>
        <DateFilterDialog
          onApply={this.hideDatePicker}
          onCancel={this.hideDatePicker}
          visible={this.state.datesPickerVisible}
        />
        <TimeFilterDialog
          onApply={this.hideTimePicker}
          onCancel={this.hideTimePicker}
          visible={this.state.timesPickerVisible}
        />
      </SafeAreaView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: filterBgColor
  },
  content: {
    paddingTop: 16,
    paddingBottom: 12
  },
  contentInterest: {
    alignItems: "center",
    flexDirection: "row",
    justifyContent: "space-between"
  },
  filterTitle: {
    color: filterShowMeTextColor,
    paddingTop: 5,
    marginRight: 8
  },
  categoriesFilterButton: {
    width: 38,
    height: 40,
    backgroundColor: categoriesFilterButtonBgColor,
    justifyContent: "center",
    alignItems: "center",
    borderTopRightRadius: 4,
    borderBottomRightRadius: 4
  },
  interestButton: {
    flex: 1,
    height: 40,
    backgroundColor: interestButtonBgColor,
    borderRadius: 4,
    justifyContent: "space-between",
    flexDirection: "row",
    alignItems: "center",
    paddingLeft: 8
  },
  interestButtonText: {
    color: interestButtonTextColor
  },
  mapButton: {
    marginLeft: 12,
    width: 52,
    height: 52,
    backgroundColor: interestButtonBgColor,
    alignItems: "center",
    justifyContent: "flex-end",
    borderRadius: 25
  },
  mapButtonText: {
    color: interestButtonTextColor,
    fontFamily: "Poppins-Bold",
    fontSize: 14,
    paddingBottom: 6
  },
  contentFilters: {
    flexDirection: "row",
    marginTop: 8
  },
  filterButton: {
    marginRight: 8
  }
});

export default FilterHeader;
