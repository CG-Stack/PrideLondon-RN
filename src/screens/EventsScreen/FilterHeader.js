// @flow
import React from "react";
import { View, StyleSheet } from "react-native";
import type { DateRange } from "../../data/date-time";
import type { EventCategoryName } from "../../data/event";
import FilterHeaderButton from "./FilterHeaderButton";
import ContentPadding from "../../components/ContentPadding";
import FilterHeaderCategories from "./FilterHeaderCategories";
import {
  filterBgColor,
  filterButtonsBgColor,
  whiteColor
} from "../../constants/colors";
import text from "../../constants/text";
import { formatDateRange } from "../../data/formatters";

export type Props = {
  onFilterCategoriesPress: Function,
  dateFilter: ?DateRange,
  onFilterButtonPress: () => void,
  onDateFilterButtonPress: () => void,
  selectedCategories: Set<EventCategoryName>,
  numTagFiltersSelected: number,
  resetAllFiltersPress: () => void
};

class FilterHeader extends React.PureComponent<Props> {
  static defaultProps = {
    resetAllFiltersPress: () => {},
    selectedCategories: new Set()
  };

  render() {
    const {
      dateFilter,
      onFilterCategoriesPress,
      selectedCategories,
      onFilterButtonPress,
      onDateFilterButtonPress,
      numTagFiltersSelected,
      resetAllFiltersPress
    } = this.props;
    const formattedDateFilter = dateFilter
      ? formatDateRange(dateFilter)
      : text.selectDates;

    const anyAppliedFilters: boolean =
      !!dateFilter || numTagFiltersSelected > 0 || selectedCategories.size > 0;

    return (
      <View accessibilityTraits={["header"]} style={styles.container}>
        <ContentPadding>
          <View>
            {anyAppliedFilters && (
              <View style={styles.clearAllWrapper}>
                <FilterHeaderButton
                  active={false}
                  text="Reset all filters"
                  label="Reset all filters"
                  style={styles.clearAll}
                  onPress={resetAllFiltersPress}
                />
              </View>
            )}
            <View testID="event-filter-header" style={styles.content}>
              <FilterHeaderCategories
                onFilterPress={onFilterCategoriesPress}
                selectedCategories={selectedCategories}
              />
            </View>
          </View>
        </ContentPadding>
        <View style={styles.contentFilters}>
          <FilterHeaderButton
            active={!!dateFilter}
            text={formattedDateFilter}
            label={`filter by date: ${formattedDateFilter}`}
            onPress={onDateFilterButtonPress}
            style={styles.filterButton}
          />
          <View style={styles.dividerLine} />
          <FilterHeaderButton
            active={numTagFiltersSelected > 0}
            text={numTagFiltersSelected > 0 ? text.filters : text.addFilters}
            label={numTagFiltersSelected > 0 ? text.filters : text.addFilters}
            onPress={onFilterButtonPress}
            style={styles.filterButton}
            badgeValue={
              numTagFiltersSelected > 0 ? numTagFiltersSelected : null
            }
          />
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: filterBgColor
  },
  content: {
    paddingTop: 16,
    paddingBottom: 12
  },
  contentFilters: {
    flexDirection: "row",
    justifyContent: "space-between",
    backgroundColor: filterButtonsBgColor,
    height: 48
  },
  filterButton: {
    flex: 1
  },
  dividerLine: {
    borderLeftWidth: StyleSheet.hairlineWidth,
    borderColor: whiteColor,
    opacity: 0.4
  },
  clearAll: {
    minHeight: 0,
    paddingTop: 16
  },
  clearAllWrapper: {
    flexDirection: "row",
    justifyContent: "flex-end"
  }
});

export default FilterHeader;
