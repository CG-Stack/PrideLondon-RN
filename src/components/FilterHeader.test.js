// @flow
import React from "react";
import { shallow } from "enzyme";
import FilterHeader from "./FilterHeader";

it("renders correctly", () => {
  const output = shallow(
    <FilterHeader dateFilter={null} timeFilter={["morning"]} />
  );
  expect(output).toMatchSnapshot();
});
