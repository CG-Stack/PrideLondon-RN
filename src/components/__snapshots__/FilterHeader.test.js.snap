// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with any date and any time (empty time set) 1`] = `
<withOrientation
  forceInset={
    Object {
      "top": "always",
    }
  }
  style={
    Object {
      "backgroundColor": "rgb(45, 47, 127)",
    }
  }
>
  <StatusBar
    animated={true}
    backgroundColor="#2d2f7f"
    barStyle="light-content"
    showHideTransition="fade"
  />
  <ContentPadding
    padding={Object {}}
    style={Object {}}
  >
    <View
      style={
        Object {
          "paddingBottom": 12,
          "paddingTop": 16,
        }
      }
      testID="filter-header"
    >
      <FilterHeaderCategories
        onFilterPress={[Function]}
        selectedCategories={Set {}}
      />
    </View>
  </ContentPadding>
  <ContentPadding
    padding={Object {}}
    style={
      Object {
        "backgroundColor": "#20215a",
        "flexDirection": "row",
        "height": 48,
        "justifyContent": "space-between",
      }
    }
  >
    <FilterHeaderButton
      active={false}
      badgeValue={null}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      text="Select dates"
    />
    <View
      style={
        Object {
          "borderColor": "#ffffff",
          "borderLeftWidth": 0.5,
        }
      }
    />
    <FilterHeaderButton
      active={false}
      badgeValue={null}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      text="Add filters"
    />
  </ContentPadding>
  <Connect(DateRangePickerDialog)
    onApply={[Function]}
    onCancel={[Function]}
    visible={false}
  />
</withOrientation>
`;

exports[`renders correctly with any date and any time 1`] = `
<withOrientation
  forceInset={
    Object {
      "top": "always",
    }
  }
  style={
    Object {
      "backgroundColor": "rgb(45, 47, 127)",
    }
  }
>
  <StatusBar
    animated={true}
    backgroundColor="#2d2f7f"
    barStyle="light-content"
    showHideTransition="fade"
  />
  <ContentPadding
    padding={Object {}}
    style={Object {}}
  >
    <View
      style={
        Object {
          "paddingBottom": 12,
          "paddingTop": 16,
        }
      }
      testID="filter-header"
    >
      <FilterHeaderCategories
        onFilterPress={[Function]}
        selectedCategories={Set {}}
      />
    </View>
  </ContentPadding>
  <ContentPadding
    padding={Object {}}
    style={
      Object {
        "backgroundColor": "#20215a",
        "flexDirection": "row",
        "height": 48,
        "justifyContent": "space-between",
      }
    }
  >
    <FilterHeaderButton
      active={false}
      badgeValue={null}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      text="Select dates"
    />
    <View
      style={
        Object {
          "borderColor": "#ffffff",
          "borderLeftWidth": 0.5,
        }
      }
    />
    <FilterHeaderButton
      active={false}
      badgeValue={null}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      text="Add filters"
    />
  </ContentPadding>
  <Connect(DateRangePickerDialog)
    onApply={[Function]}
    onCancel={[Function]}
    visible={false}
  />
</withOrientation>
`;

exports[`renders correctly with date range and two times 1`] = `
<withOrientation
  forceInset={
    Object {
      "top": "always",
    }
  }
  style={
    Object {
      "backgroundColor": "rgb(45, 47, 127)",
    }
  }
>
  <StatusBar
    animated={true}
    backgroundColor="#2d2f7f"
    barStyle="light-content"
    showHideTransition="fade"
  />
  <ContentPadding
    padding={Object {}}
    style={Object {}}
  >
    <View
      style={
        Object {
          "paddingBottom": 12,
          "paddingTop": 16,
        }
      }
      testID="filter-header"
    >
      <FilterHeaderCategories
        onFilterPress={[Function]}
        selectedCategories={Set {}}
      />
    </View>
  </ContentPadding>
  <ContentPadding
    padding={Object {}}
    style={
      Object {
        "backgroundColor": "#20215a",
        "flexDirection": "row",
        "height": 48,
        "justifyContent": "space-between",
      }
    }
  >
    <FilterHeaderButton
      active={true}
      badgeValue={null}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      text="2 Feb - 3 Feb"
    />
    <View
      style={
        Object {
          "borderColor": "#ffffff",
          "borderLeftWidth": 0.5,
        }
      }
    />
    <FilterHeaderButton
      active={false}
      badgeValue={null}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      text="Add filters"
    />
  </ContentPadding>
  <Connect(DateRangePickerDialog)
    onApply={[Function]}
    onCancel={[Function]}
    visible={false}
  />
</withOrientation>
`;

exports[`renders correctly with single date and single time 1`] = `
<withOrientation
  forceInset={
    Object {
      "top": "always",
    }
  }
  style={
    Object {
      "backgroundColor": "rgb(45, 47, 127)",
    }
  }
>
  <StatusBar
    animated={true}
    backgroundColor="#2d2f7f"
    barStyle="light-content"
    showHideTransition="fade"
  />
  <ContentPadding
    padding={Object {}}
    style={Object {}}
  >
    <View
      style={
        Object {
          "paddingBottom": 12,
          "paddingTop": 16,
        }
      }
      testID="filter-header"
    >
      <FilterHeaderCategories
        onFilterPress={[Function]}
        selectedCategories={Set {}}
      />
    </View>
  </ContentPadding>
  <ContentPadding
    padding={Object {}}
    style={
      Object {
        "backgroundColor": "#20215a",
        "flexDirection": "row",
        "height": 48,
        "justifyContent": "space-between",
      }
    }
  >
    <FilterHeaderButton
      active={true}
      badgeValue={null}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      text="2 Feb"
    />
    <View
      style={
        Object {
          "borderColor": "#ffffff",
          "borderLeftWidth": 0.5,
        }
      }
    />
    <FilterHeaderButton
      active={false}
      badgeValue={null}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      text="Add filters"
    />
  </ContentPadding>
  <Connect(DateRangePickerDialog)
    onApply={[Function]}
    onCancel={[Function]}
    visible={false}
  />
</withOrientation>
`;

exports[`renders correctly with tag filters selected 1`] = `
<withOrientation
  forceInset={
    Object {
      "top": "always",
    }
  }
  style={
    Object {
      "backgroundColor": "rgb(45, 47, 127)",
    }
  }
>
  <StatusBar
    animated={true}
    backgroundColor="#2d2f7f"
    barStyle="light-content"
    showHideTransition="fade"
  />
  <ContentPadding
    padding={Object {}}
    style={Object {}}
  >
    <View
      style={
        Object {
          "paddingBottom": 12,
          "paddingTop": 16,
        }
      }
      testID="filter-header"
    >
      <FilterHeaderCategories
        onFilterPress={[Function]}
        selectedCategories={Set {}}
      />
    </View>
  </ContentPadding>
  <ContentPadding
    padding={Object {}}
    style={
      Object {
        "backgroundColor": "#20215a",
        "flexDirection": "row",
        "height": 48,
        "justifyContent": "space-between",
      }
    }
  >
    <FilterHeaderButton
      active={false}
      badgeValue={null}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      text="Select dates"
    />
    <View
      style={
        Object {
          "borderColor": "#ffffff",
          "borderLeftWidth": 0.5,
        }
      }
    />
    <FilterHeaderButton
      active={true}
      badgeValue={2}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      text="Filters"
    />
  </ContentPadding>
  <Connect(DateRangePickerDialog)
    onApply={[Function]}
    onCancel={[Function]}
    visible={false}
  />
</withOrientation>
`;
